templates: grants_table_scripts: Apply conditional responsive classes

codemacabre opened the pull request:
To make Datatables' built-in responsive algorithms work a bit better, this change checks if the page / iframe URL includes 'widget' to detect if the table is within a widget. If not, it applies different classes to allow breakpoints to be a bit wider to accommodate the sidebar in the main search view.

Fixes: https://github.com/ThreeSixtyGiving/grantnav/issues/1009

michaelwood commented on grantnav/frontend/templates/components/grants_table_scripts.html:
```diff
@@ -27,6 +27,7 @@
 const params = new URLSearchParams(window.location.search);
 const inputRows = params.get('rows');
 const inputColumns = params.getAll('widgetColumn');
+const isWidget = window.location.href.includes('widget');
```
And if you search for "widget" in grantnav search :) probably need to use window.location.pathname ?

michaelwood reviewed (changes requested):
Everything else looks good, I tested it on my local machine

codemacabre committed:
templates: grants_table_scripts: Apply conditional responsive classes

Checks if the URL includes 'widget' and applies different classes if the table is within a widget

Fixes: https://github.com/ThreeSixtyGiving/grantnav/issues/1009
